keps/sig-network/3698-multi-network/README.md

```diff
@@ -0,0 +1,363 @@
+networking model, extending their functionality.
+   * We will introduce an PodNetwork API object that represents the existing
+     infrastructure’s networking.
+3. PodNetwork functions as a handle that decouples the implementation
+(i.e. infrastructure/network admins) from the consumer (cluster user). This
+abstraction follows the Kubernetes model of decoupling intent from implementation.
+4. PodNetwork shall not define any implementation specific parameters in its
+specification.
+5. PodNetwork shall provide cluster generic options only.
+6. PodNetwork can reference to implementation-specific parameters.
+7. PodNetwork is the object Application developer will reference in their
+workflows (e.g. Pod, Service).
+8. Cluster Default PodNetwork is the PodNetwork the cluster has been created
+with initially.
+9. Cluster Default PodNetwork cannot be removed, but can be replaced.
+10. Pods can reference one or more PodNetworks when trying to attach to them.
+    * This list is explicit and all the PodNetworks that has to be available in
+      the Pod has to be listed.
+11. Workloads access to PodNetwork is controlled via RBAC configuration.
+12. When Pods does not specify any reference to PodNetwork it connects to Cluster
+Default PodNetwork (network the cluster has been created with).
+13. Pod shall be able to provide additional configuration on how it attaches to
+a PodNetwork.
+14. Every Pod connected to specific PodNetwork must have connectivity within
+that network across the Cluster.
+15. A Pod connected to specific PodNetwork may or may not have cross 
+connectivity between different PodNetworks.
+16. Pods attached to PodNetworks are connected to each other in a manner 
+defined by the PodNetwork implementation.
+17. Implementation-agnostic PodNetwork Interface information for each attachment
+will be exposed to runtime Pod (via e.g. environment variables, downward API etc.).
+
+#### Phase II (scheduler, kubelet and API probing for PodNetwork)
+18. PodNetworks can be selectively (per Node) available in the Cluster. This
+does NOT apply to the Cluster Default PodNetwork.
+19. Kubelet network-based probing can be optionally enabled for the non- Cluster
+Default PodNetwork attachments to Pod.
+20. Kubernetes API can be optionally accessible via the non- Cluster Default
+PodNetwork attachments to Pod.
+21. Kubernetes API can optionally reach the Pod via the non- Cluster Default
+PodNetwork attachments .
+
+#### Phase III (basic Kubernetes features integration)
+22. All PodNetwork attachments to Pod are optionally able to provide Service
+and NetworkPolicy functionality.
+
+#### Phase IV (extended functionality)
+23. Have capability to override Cluster Default PodNetwork in the namespace scope.
+24. Have capability to add/remove attachments to/from running Pods.
+25. Have capability to add/remove IP on running Pods network interfaces.
+
+### Risks and Mitigations
+
+N/A
+
+## Design Details
+
+N/A
```

Comment:
here you need to describe in detail the APIs and the behaviors

Reply:
Phase I coming soon